sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -3625,6 +3625,20 @@ __ESIMD_API
   scatter<T, N>(acc, ByteOffsets, vals, glob_offset, mask);
 }
 
+#ifdef __ESIMD_FORCE_STATELESS_MEM
+template <typename T, int N, typename AccessorTy, typename Toffset>
+__ESIMD_API std::enable_if_t<
+    (detail::isPowerOf2(N, 32)) &&
```

Comment:
Do we really need this condition for `isPowerOf2` here? In stateless mode it will eventually reach (usm-sc-1), which can handle non-power-of-2 lengths and lower it to LSC.

Reply:
removed in https://github.com/intel/llvm/pull/12670/commits/c0ff4b7f8a89607245cee65423723ec538430f8e